addons/notes/src/register.tsx

```diff
@@ -1,18 +1,44 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
 import addons from '@storybook/addons';
 
 import styled from '@emotion/styled';
 
+interface NotesChannel {
+  on: (listener: string, callback: (text: string) => void) => any; // todo check correct return value definition
+  emit: any; // todo check correct definition
+  removeListener: (listener: string, callback: (text: string) => void) => void;
+}
+
+interface NotesApi {
+  onStory: (callback: () => void) => void;
+  getQueryParam: any; // todo check correct definition
+  setQueryParams: any; // todo check correct definition
+}
+
+interface NotesProps {
+  active: boolean;
+  channel: NotesChannel;
+  api: NotesApi;
+}
+
+interface NotesState {
+  text: string;
+}
+
 const Panel = styled.div({
   padding: 10,
   boxSizing: 'border-box',
   width: '100%',
 });
 
-export class Notes extends React.Component {
-  constructor(...args) {
-    super(...args);
+export class Notes extends React.Component<NotesProps, NotesState> {
+
+  // todo check how react typescript does this, especially unmounted
+  stopListeningOnStory: any; // todo check correct definition
+  unmounted: any; // todo check correct definition
+
+  constructor(...args: any) {
+    super(args /* todo: This was ...args before, check if this was a bug */);
     this.state = { text: '' };
```

Comment:
There is no need to define the state in the constructor with React, or bind methods. You can just write it as a property:
```
public state: NotesState = { text: '' };
// ...
private onAddNotes = (args) => { /* ... */ }
```
instead of:
```
constructor(...args: any[]) {
   this.state = { text: '' };
   this.onAddNotes = this.onAddNotes.bind(this);
}
// ...
onAddNotes(args) { /* ... */ }
```
As result - usually there is no need to define constructor explicitly